rasterio/session.py

```diff
@@ -316,6 +316,38 @@ def get_credential_options(self):
 
         """
         return {k.upper(): v for k, v in self.credentials.items()}
+    
+
+class GSSession(OSSSession):
+    """Configures access to secured resources stored in Google Cloud Storage
+    """
+    def __init__(self, google_application_credentials):
```

Comment:
```suggestion
    def __init__(self, google_application_credentials=None):
```

@theduckylittle if we let this default to `None` we could let GDAL pick up the 7th step in https://www.gdal.org/gdal_virtual_file_systems.html#gdal_virtual_file_systems_vsigs.